mne/export/_edf.py

```diff
@@ -0,0 +1,212 @@
+    - if Info stores techniciain information, allow writing here.
+    """
+    phys_dims = 'uV'
+
+    if fmt == 'bdf':
+        digital_min = -8388607
+        digital_max = 8388607
+    elif fmt == 'edf':
+        digital_min = -32767
+        digital_max = 32767
+
+    # load data first
+    raw.load_data()
+
+    # remove extra epoc and STI channels
+    drop_chs = ['epoc']
+    if not (raw.filenames[0].endswith('.fif')):
+        drop_chs.append('STI 014')
+
+    ch_names = [ch for ch in raw.ch_names if ch not in drop_chs]
+    n_chs = len(ch_names)
+    n_times = raw.n_times
+    sfreq = int(raw.info['sfreq'])
+    n_secs = n_times / sfreq
+
+    # get any filter information applied to the data
+    lowpass = raw.info['lowpass']
+    highpass = raw.info['highpass']
+    linefreq = raw.info['line_freq']
+    filter_str_info = f"HP:{highpass}Hz LP:{lowpass}Hz N:{linefreq}Hz"
+
+    # get data in uV
+    # XXX: what about non-EEG data?... How to handle?
+    units = dict()
+    units['eeg'] = 'uV'
+    units['ecog'] = 'uV'
+    units['seeg'] = 'uV'
+
+    # get the entire dataset in uV
+    data = raw.get_data(units=units, picks=ch_names)
+
+    if physical_range == 'auto':
+        # get max and min for each channel type data
+        ch_types_phys_max = dict()
+        ch_types_phys_min = dict()
+
+        for idx, ch_name in enumerate(raw.ch_names):
+            ch_type = raw.get_channel_types(picks=ch_name)[0]
+
+            ch_type_data = data[idx, :]
+            if ch_type not in ch_types_phys_max:
+                ch_types_phys_max[ch_type] = ch_type_data.max()
+                ch_types_phys_min[ch_type] = ch_type_data.min()
+            else:
+                ch_types_phys_max[ch_type] = max(
+                    ch_type_data.max(), ch_types_phys_max[ch_type])
+                ch_types_phys_min[ch_type] = min(
+                    ch_type_data.min(), ch_types_phys_min[ch_type])
+    else:
+        # get the physical min and max of the data in uV
+        # Physical ranges of the data in uV is usually set by the manufacturer
+        # and properties of the electrode. In general, physical max and min
+        # should be the clipping levels of the ADC input and they should be
+        # the same for all channels. For example, Nihon Kohden uses +3200 uV
+        # and -3200 uV for all EEG channels (which are the actual clipping
+        # levels of their input amplifiers & ADC).
+        # For full discussion, see: https://github.com/sccn/eeglab/issues/246
+        pmin, pmax = -3200, 3200
+
+        # check that physical min and max is not exceeded
+        if data.max() > pmax:
+            raise RuntimeError(f'The maximum uV of the data {data.max()} '
+                               f'is more then physical max passed in {pmax}.')
+        if data.min() < pmin:
+            raise RuntimeError('The minimum uV of the data {data.min()} '
+                               f'is less then physical min passed in {pmin}.')
```

Comment:
it looks like you're ignoring `physical_range` when it's not `"auto"` and hard-setting the values, instead of using the tuple passed by the user.

Reply:
why was this marked "resolved"?  You're hard-setting `pmin, pmax` on line 104 and then on lines 108 and 111 talking about the value the user "passed in".  The function doesn't do what is documented w/r/t passing a tuple to `physical_range`